File: frontend/src/models/ExtensionSocial.ts
export interface ExtensionSocial {
  id: number;
  periodo: string;
  conferencias_dictadas: number;
  cursos_ofrecidos: number;
  diplomados_ofrecidos: number;
  talleres_ofrecidos: number;
  consultorias: number;
  asistentes: number;
  horas_ofrecidas: number;
  participacion_estudiantil: number;
  participacion_egresados: number;
  participacion_profesores: number;
  ingreso_neto: number;
}

export interface ExtensionSocialStats {
  total_registros: number;
  total_conferencias: number;
  total_cursos: number;
  total_diplomados: number;
  total_talleres: number;
  total_consultorias: number;
  total_asistentes: number;
  total_horas: number;
  total_participacion_estudiantil: number;
  total_participacion_egresados: number;
  total_participacion_profesores: number;
  total_ingreso_neto: number;
  promedio_ingreso_neto: number;
}

// Chart data types
export interface ActivitiesChartData {
  periodo: string;
  conferencias: number;
  cursos: number;
  diplomados: number;
  talleres: number;
  consultorias: number;
}

export interface EvolutionChartData {
  periodo: string;
  asistentes: number;
  ingreso_neto: number;
}

export interface ParticipationChartData {
  name: string;
  value: number;
}

File: backend/app/schemas/extension_social.py
from pydantic import BaseModel


class ExtensionSocialResponse(BaseModel):
    id: int
    periodo: str
    conferencias_dictadas: int
    cursos_ofrecidos: int
    diplomados_ofrecidos: int
    talleres_ofrecidos: int
    consultorias: int
    asistentes: int
    horas_ofrecidas: int
    participacion_estudiantil: int
    participacion_egresados: int
    participacion_profesores: int
    ingreso_neto: float

    class Config:
        from_attributes = True


class ExtensionSocialStatsResponse(BaseModel):
    total_registros: int
    total_conferencias: int
    total_cursos: int
    total_diplomados: int
    total_talleres: int
    total_consultorias: int
    total_asistentes: int
    total_horas: int
    total_participacion_estudiantil: int
    total_participacion_egresados: int
    total_participacion_profesores: int
    total_ingreso_neto: float
    promedio_ingreso_neto: float
